### PR TITLE
CRM-19549 : New pledge data entry page shows hard coded $ for installments

### PR DESCRIPTION
**Pull request by jitendrapurohit**
* [CRM-19549: New pledge data entry page shows hard coded $ for installments](https://issues.civicrm.org/jira/browse/CRM-19549)

### CODE REVIEW

**Comment by jitendrapurohit** [comment]
jenkin, retest this please

File: templates/CRM/Pledge/Form/Pledge.tpl [hunk 1]
```diff
@@ -147,10 +154,23 @@ cj(document).ready( function() {
...
+
+function replaceCurrency(val) {
+  var eachPaymentAmout = cj('#eachPaymentAmount');
+  var pos = val.indexOf("(") + 1;
+  if (pos) {
+    val = val.slice(pos, val.lastIndexOf(")"));
+  }
+  cj('.currency-symbol').text(val).append("&nbsp;").append(eachPaymentAmout);
+}
+
```
Comment (pradpnayak) [inline templates/CRM/Pledge/Form/Pledge.tpl]: I feel this is an outdated pattern.  We no longer use cj. 
The way to update this code would be to wrap the whole thing in CRM.$(function($) { ... }); and replace all cj with $.

**Comment by pradpnayak** [comment]
I tested above patch. The Currency symbol gets changed when Currency is changed from currency option field. However there are still some error's 

1. When there is error on the form, then currency is switched back to $. 
To Replicate :
Add Pledge -> Don't Select Contact (leave contact blank). Change the currency to INR (₨) ($ is replaced by (₨)) and submit the form. You will see the currency symbol is $ even though INR is selected.

2. On Edit of Pledge Form, Symbol is not shown. (regression error)

3. (Design issue) Can we format the Amount Columns
  a.  **Pledged**, **total paid**, **Balanced**, **Next amount** under Pledge tab of Contact to include currency? 
  b. Under pledge instalment for **Amount due** and **Amount Paid** column? 
  c. On Pledge View Form **Total Pledge Amount** and **To be paid** rows.

**Comment by pradpnayak** [comment]
I think #3 is covered under https://github.com/civicrm/civicrm-core/pull/9476

**Comment by jitendrapurohit** [comment]
@pradpnayak Can I assume point 2 is too covered in #9476 ? Since the currency-symbol do get displayed for me on Edit Pledge Form.

edit_pledge

**Comment by pradpnayak** [comment]
yes #2 also covered in https://github.com/civicrm/civicrm-core/pull/9476

**Comment by monishdeb** [comment]
@pradpnayak is this PR ready to merge ?

**Comment by pradpnayak** [comment]
i guess # 1 still need to be addressed

**Comment by jitendrapurohit** [comment]
@pradpnayak I see # 1 is included in the update since it works fine for me. Is there something I'm missing ? 

pledge_validation

**Comment by pradpnayak** [comment]
Sorry, i will test the PR now .

**Comment by pradpnayak** [comment]
During validation error, symbol is not shown for missing currency symbol. But the same is shown with Currency name when currency is switched.

**Comment by jitendrapurohit** [comment]
@pradpnayak As per [comment written here](https://issues.civicrm.org/jira/browse/CRM-19701?focusedCommentId=97313&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-97313), its fine to not show symbols for few currencies. I've opened a new ticket to add missing symbols for them https://issues.civicrm.org/jira/browse/CRM-19733

**Comment by pradpnayak** [comment]
@jitendrapurohit i feel then it should also hold true when we change the currency symbol. Its little confusing when there is a validation error the currency name for missing currency symbol is missing but it appears when we try to change currency from currency drop down.

If the core is happy to accept this design, then please go ahead and merge this PR, as i have tested on local and it works very well besides above mentioned issue.

**Comment by jitendrapurohit** [comment]
@pradpnayak fixed the js part to show symbol only when it is present.

**Comment by pradpnayak** [comment]
Tested, working fine

**Pull request merged**